### PR TITLE
add EightBase package.

### PR DESCRIPTION
**Pull request by vorobeez**
@sebscholl, @esemushin what were added:

- Events to Auth. You can specify in constructor `onAuthorized`, `onAuthorizeFailed`, `onRefreshed`, `onRefreshFailed`, `onSignedOut`. Or call `on` method with required event.
(https://github.com/8base/js-sdk/pull/9/files#diff-1cf5b10c87e919f81d5f1ea503f15074R99)
(https://github.com/8base/js-sdk/pull/9/files#diff-1cf5b10c87e919f81d5f1ea503f15074R35)
- Errors map to Api. 
(https://github.com/8base/js-sdk/pull/9/files#diff-3102012833924b19b9c2ac08b54ec566R272)
(https://github.com/8base/js-sdk/pull/9/files#diff-e733ba9ded6a614d46191a184bffbdecR20)
- Rerun function to a error callback to Api.
(https://github.com/8base/js-sdk/pull/9/files#diff-3102012833924b19b9c2ac08b54ec566R272)
(https://github.com/8base/js-sdk/pull/9/files#diff-e733ba9ded6a614d46191a184bffbdecR20)
- EightBase module
(https://github.com/8base/js-sdk/pull/9/files#diff-0826af772b9e1fb16ac9777b78234533R9)
- ExtendedAuth with graphql methods `signIn`, `signUp`, `signUpWithToken`.
(https://github.com/8base/js-sdk/pull/9/files#diff-8be2624e3f5548da7d40c1b0c8f3eaedR18)
- Auto token refresh enabled by default to EightBase.
(https://github.com/8base/js-sdk/pull/9/files#diff-0826af772b9e1fb16ac9777b78234533R27)

### CODE REVIEW

File: packages/8base/__tests__/autoTokenRefresh.test.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+  beforeAll(() => {
+    originalFetch = fetch;
+    // @ts-ignore
+    fetch = jest.fn();
+  });
+
+  beforeEach(() => {
+    eightBase = new EightBase({
+      authProfileId: AUTH_PROFILE_ID,
```
Comment (sebscholl) [inline packages/8base/__tests__/autoTokenRefresh.test.ts]: `authProfileId ` should be in the Auth strategy settings object, not top level of the EightBase class.

Comment (vorobeez) [reply packages/8base/__tests__/autoTokenRefresh.test.ts]: Then by that logic `workspaceId` also should be in a api settings object. Is that right?

Comment (sebscholl) [reply packages/8base/__tests__/autoTokenRefresh.test.ts]: No. Auth and API are our first 2 ask modules. We have several or more that will be soon adding, all of which will use the workspaceId.

Comment (vorobeez) [reply packages/8base/__tests__/autoTokenRefresh.test.ts]: Fixed

**Pull request merged**